### ISSUE TITLE
SPEI : can't figure it out

### ISSUE THREAD

**Issue by aulemahal** [Ouranosinc/xclim#1602]
### Discussed in https://github.com/Ouranosinc/xclim/discussions/1589

See comment : https://github.com/Ouranosinc/xclim/discussions/1589#discussioncomment-8155294 for a description of the bug.

Issue comes from this line : https://github.com/Ouranosinc/xclim/blob/bd6a6f87f9cf5897de198108b369efe5356bed87/xclim/indices/stats.py#L765

Having `freq='D'` (or `freq=None` with a daily dataset) results in the step previous to the above line grouping the data with `time.dayofyear`. Thus at the line above, the "temporal" dimension is named `dayofyear` and not `day`, which is why this fails.

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct</div>

**Comment by coxipi** [comment]
I thought I had a test for daily-SPI, apparently not. 

I've made the changes, and wanted to add tests to compare with monocongo, but the results don't match. It's a bit hard to find a 3rd party for a tie-breaker between xclim and monocongo since the daily SPI is not that common. I wanted to try with this one: https://github.com/e-baumer/standard_precip, but it only partially obtain the correct values I've established for monthly SPI where monocongo and xclim do match.
